### PR TITLE
Only check for daily double when an amount has been requested.

### PR DESCRIPTION
**Pull request by JaredReisinger**
We were seeing daily double messages even when no amount had been made by the original highfiver, which looks kind of weird.

_Other changes:_
Allow explicit zero values for daily double and boomerang; makes testing easier and more stable.
Clean up test code.
Add explicit test for daily double.

### CODE REVIEW

File: test/command_spec.coffee [hunk 1]
```diff
@@ -161,6 +169,22 @@ describe 'highfive', ->
...
+        prep done
+    afterEach ->
+        process.env.HUBOT_HIGHFIVE_DOUBLE_RATE = '0'
+        do cleanup
+
+    it 'should double the amount', (done) ->
+        message_response 'highfive @foo $10 for something', 'send', (e, strs) ->
+            return unless strs.match /gift card is on its way/
+            expect(strs).to.match /A \$20 gift card is on its way/
+            do done
```
Comment (ben) [inline test/command_spec.coffee:186]: :heart_eyes:

File: src/highfive.coffee [hunk 2]
```diff
@@ -201,14 +202,15 @@ module.exports = (robot) ->
...
-                            chatService.boomerang roomid, from_obj, 
+                            chatService.boomerang roomid, from_obj,
                             sheet.logToSheet robot, [
-                                m,
-                                from_obj.email,
-                                from_obj.email,
-                                10,
-                                '(Boomerang)',
-                                order.reward.number,
+                                m,                    # date (why not *new* order date?)
```
Comment (ben) [inline src/highfive.coffee:207]: To answer your question: no good reason. I just didn't think about recording a different time. Having them the same is nice for tracking that the original and the boomerang are related, but a few seconds difference probably wouldn't matter.

**Comment by ben** [comment]
I love it. Did you want to fix the timestamp generation, or should I just merge this?

**Comment by JaredReisinger** [comment]
Go ahead and merge it.  I can submit the timestamp as a second PR.  Thanks!

**Pull request merged**